yarn-project/pxe/src/pxe_oracle_interface/pxe_oracle_interface.test.ts

```diff
@@ -563,6 +565,109 @@ describe('PXEOracleInterface', () => {
+      expect(result?.uniqueNoteHashesInTx).toEqual(txEffect.noteHashes);
+      expect(result?.txHash).toEqual(scopedLog.txHash);
+      expect(result?.firstNullifierInTx).toEqual(txEffect.nullifiers[0]);
+
+      expect(aztecNode.getLogsByTags).toHaveBeenCalledWith([tag]);
+      expect(aztecNode.getTxEffect).toHaveBeenCalledWith(scopedLog.txHash);
+    });
+
+    it('throws if multiple logs found for tag', async () => {
+      aztecNode.getLogsByTags.mockResolvedValue([[scopedLog, scopedLog]]);
+
+      await expect(pxeOracleInterface.getLogByTag(tag)).rejects.toThrow(/Got 2 logs for tag/);
+    });
+
+    it('throws if tx effect not found', async () => {
+      aztecNode.getLogsByTags.mockResolvedValue([[scopedLog]]);
+      aztecNode.getTxEffect.mockResolvedValue(undefined);
+
+      await expect(pxeOracleInterface.getLogByTag(tag)).rejects.toThrow(/failed to retrieve tx effects/);
+    });
+
+    it('trims zero padding from log fields', async () => {
+      const logContent = [Fr.random(), Fr.random()];
+      const logContentWithPadding = [
+        ...logContent,
+        ...Array(PRIVATE_LOG_SIZE_IN_FIELDS - logContent.length).fill(Fr.ZERO),
+      ] as Tuple<Fr, typeof PRIVATE_LOG_SIZE_IN_FIELDS>;
+
+      const scopedLogWithPadding = new TxScopedL2Log(
+        TxHash.random(),
+        randomInt(100),
+        randomInt(100),
+        randomInt(100),
+        new PrivateLog(logContentWithPadding),
+      );
+
+      aztecNode.getLogsByTags.mockResolvedValue([[scopedLogWithPadding]]);
+      aztecNode.getTxEffect.mockResolvedValue(randomInBlock(txEffect));
+
+      const result = await pxeOracleInterface.getLogByTag(tag);
+
+      expect(result?.logContent).toEqual(logContent);
+    });
+  });
+
+  describe('removeNullifiedNotes', () => {
+    it('should remove nullified notes', async () => {
+      const noteDao = await NoteDao.random();
+      const recipient = noteDao.recipient;
+      const nullifierIndex = randomInBlock(123n);
+
+      jest.spyOn(keyStore, 'getAccounts').mockResolvedValue([recipient]);
+      jest.spyOn(noteDataProvider, 'getNotes').mockResolvedValue([noteDao]);
+      jest.spyOn(aztecNode, 'findLeavesIndexes').mockResolvedValue([nullifierIndex]);
+      jest.spyOn(noteDataProvider, 'removeNullifiedNotes').mockResolvedValue([noteDao]);
+
+      await pxeOracleInterface.removeNullifiedNotes(contractAddress);
+
+      expect(keyStore.getAccounts).toHaveBeenCalled();
+      expect(noteDataProvider.getNotes).toHaveBeenCalledWith({ contractAddress, recipient });
+      expect(aztecNode.findLeavesIndexes).toHaveBeenCalledWith('latest', MerkleTreeId.NULLIFIER_TREE, [
+        noteDao.siloedNullifier,
+      ]);
+      expect(noteDataProvider.removeNullifiedNotes).toHaveBeenCalledWith(
+        [
+          {
+            data: noteDao.siloedNullifier,
+            l2BlockNumber: nullifierIndex.l2BlockNumber,
+            l2BlockHash: nullifierIndex.l2BlockHash,
+          },
+        ],
+        recipient,
```

Comment:
I'm typically not a fan of this kind of spy tests because they seem a bit brittle and more of a copy of the implementation - we're just checking that the things the fn calls are being called. But I also don't have a lot of experience with writing tests for this kind of components, so unsure as to whether my concern is real. 

I'd rather see the tests more deal with the setup and results, and not so much what the fn actually did, which seems more like impl details.

Reply:
You have a point. Re-worked the tests but decided to keep the `removeNullifiedNotes` spy on note data provider as that one I think makes sense as it protects against test setup issues (was the note really correctly not removed or did I just fail to correctly setup accounts and hence the code was not triggered at all?).